sdk/go/common/workspace/project.go

```diff
@@ -624,13 +624,13 @@ func (e *Environment) Imports() []string {
 	if err != nil || len(diags) != 0 || def == nil {
 		return nil
 	}
-	names := fx.ToSet(fx.Map(fx.IterSlice(def.Imports.GetElements()), func(imp *ast.ImportDecl) string {
+	names := fx.ToSlice(fx.Map(fx.IterSlice(def.Imports.GetElements()), func(imp *ast.ImportDecl) string {
 		return imp.Environment.GetValue()
 	}))
 	if len(def.Values.GetEntries()) != 0 {
-		names.Add("yaml")
+		names = append(names, "yaml")
 	}
-	return fx.ToSlice(fx.IterSet(names))
+	return names
```

Comment:
I had to change this because the previous version of this function did not preserve the order of the imports and also removed duplicates.

I think it is more correct to show the imports as listed (including any duplicates if they exist) since the import order determines merge behavior and removing duplicates could obfuscate the behavior.

Reply:
This sounds reasonable to me!

Reply:
Yeah, this should be fine assuming we don't depend on the deduping elsewhere. If we do, the callee can always dedup.

Reply:
Makes sense. I think I added it b/c I had some changes that used it to attach imported envs to update metadata